src/main/java/org/semver4j/internal/Coerce.java

```diff
@@ -4,30 +4,30 @@
 import org.jetbrains.annotations.Nullable;
 
 import java.util.Locale;
+import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 import static java.lang.String.format;
 import static java.util.regex.Pattern.compile;
-import static org.semver4j.internal.Tokenizers.COERCE;
 
 public class Coerce {
     @NotNull
-    private static final Pattern pattern = compile(COERCE);
+    private static final Pattern PATTERN = compile(
+        "(^|\\D)(\\d{1,16})(?:\\.(\\d{1,16}))?(?:\\.(\\d{1,16}))?(?:$|\\D)"
```

Comment:
If it's only used in one location, my thought was to keep it in that location instead of having to jump from one file to another. Also, injecting the string directly enabled the IDE to highlight the regular expression which identified the changes to groups 1 and 5 to use `\\D` instead of `[^\\d]`. I could move it back, but at least the `Pattern` instance should be moved over to `Tokenizers` as well so that the IDE highlighting can be preserved.

Reply:
I don't fully feel that change, but if you strongly feel it, let's do this.